### ISSUE TITLE
Storage and Presentation of Accessibility Check Results

### ISSUE THREAD

**Issue by ajkiessl** [psu-libraries/scholarsphere#1613]
There are several parts to this, so this may need to be an epic of it's own.

**Data model:** A new data model (maybe `AccessibilityCheckResult`) that `belongs_to` a `FileResource` and a `has_one` relationship from `FileResource` to `AccessibilityCheckResult`.  There shouldn't be more than one accessibility check for a `FileResource`.   Alternatively, we could store the results of the accessibility check with FileResource.

The `AccessibilityCheckResult` should store whether the check passed or failed, and the contents of the report.  Since the report is in JSON, we could try storing the report as json-b in the database.  These both should be required fields if using a separate data model.

~~**Parsing the data from the report:** This depends on how #1602 is implemented.  We could pull our file from S3, store it temporarily, then pass it to Adobe to be checked.  Adobe will send us a JSON response with the report.  Another, option is to send a presigned URL to Adobe, they download the file to be checked, and they send us a zip file containing the original file and the JSON report.  How we parse will depend on which implementation is chosen.~~  We'll go with the solution that doesn't involve presigned urls.  Since the parsing can be done in #1602, we don't need to worry about this in this issue.

**Presentation:** The UI on the publish page already lists the uploaded files.  We could add another column in that list that gives the pass/fail outcome of the accessibility check.  There will need to be some polling to check the database for when the `AccessibilityCheckResult` is finally stored.  We'll also want a link to download the report.  The report should be human readable, not JSON.  If any of the files fails the check, there should be a link at the end of the list of files directing the user to a site for help on remediation.  We'll need to get the link for this site.

We'll need a place in the "Work Settings" page that lists the files and a link to download the reports.  We want both edit users and admins to be able to access these.

**Comment by Smullz622** [comment]
The result column added to the table should have "_ out of _ passed" if the checker has completed, 'Processing' if the checker is still running, and 'Needs manual review' for non pdf files that can't run through the checker (or perhaps if there's an error in the checker & it can't complete). This will need to be dynamic (consider hotwire/stimulus) & give real time updates since we are blocking publish until the checker finishes.
